tests/t0114_gateway_subdomains_test.go

```diff
@@ -58,5 +56,7 @@ func TestGatewaySubdomains(t *testing.T) {
 		},
 	}
 
-	Run(t, tests)
+	if SubdomainGateway.IsEnabled() {
+		Run(t, tests)
+	}
```

Comment:
I added it like this for now for simplicity. What I imagine in the future is that each test case comes with a list of specs it tests. Before running the test case, we check if all the specs are enabled. If not, we skip that test case. Additionally, if a test case fails, we could link to the spec in the error message.

To not have to specify specs for each test case, we could wrap the encapsulating array in some class and store the defaults there.